src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -298,15 +296,13 @@ public IEnumerable<DiagnosticData> ConvertToLocalDiagnostics(Document targetDocu
                 foreach (var analyzer in existingAnalyzers)
                 {
                     if (existing.TryGetValue(analyzer, out var analysisResult) &&
-                        analysisResult.Version == version &&
-                        !analyzer.IsOpenFileOnly(project.Solution.Workspace))
```

Comment:
hmmm.. we might filter out all hidden analyzer for closed file one, so this basically assume it will never be here I guess.